icu4c/source/test/fuzzer/number_format_fuzzer.cc

```diff
@@ -13,17 +13,31 @@ IcuEnvironment* env = new IcuEnvironment();
 
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
   UErrorCode status = U_ZERO_ERROR;
+  uint16_t rnd = 0;
 
-  auto rng = CreateRng(data, size);
-  const icu::Locale& locale = GetRandomLocale(&rng);
+  if (size < 2) {
+    return 0;
+  }
+
+  rnd = *(reinterpret_cast<const uint16_t *>(data));
+  data = data + 2;
+  size = size - 2;
+
+  size_t unistr_size = size/2;
+  std::unique_ptr<char16_t[]> fuzzbuff(new char16_t[unistr_size]);
+  std::memcpy(fuzzbuff.get(), data, unistr_size * 2);
+
+  const icu::Locale& locale = GetRandomLocale(rnd);
 
   std::unique_ptr<icu::NumberFormat> fmt(
       icu::NumberFormat::createInstance(locale, status));
-  if (U_FAILURE(status)) return 0;
+  if (U_FAILURE(status)) {
+    return 0;
+  }
 
-  icu::UnicodeString str(UnicodeStringFromUtf8(data, size));
+  icu::UnicodeString fuzzstr(fuzzbuff.get(), unistr_size);
```

Comment:
Use the aliasing UnicodeString constructor.

Reply:
Aliasing constructor used.